### PR TITLE
adding my keyboard

### PR DESCRIPTION
**Pull request by mechmehmet**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/2d4d/readme.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# 2d4d
+
+https://i.imgur.com/famGFG8.jpg
+
+https://i.imgur.com/IfoQUnc.jpg
+
+https://imgur.com/a/e9hCvfe
+
+*Handwired %50 keyboard (48 switches), you can use 1 or 2 encoders instead of switch and vise versa*
```
Comment (zvecr) [inline keyboards/2d4d/readme.md:9]: Needs to be in the `keyboards/handwired` folder.

File: keyboards/2d4d/2d4d.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+) { \
+    { K000, K001, K002, K003, K004, K005 }, \
+    { K100, K101, K102, K103, K104, K105 }, \
+    { K200, K201, K202, K203, K204, K205 }, \
+    { K300, K301, K302, K303, K304, K305 }, \
+    { K400, K401, K402, K403, K404, K405 }, \
+    { K500, K501, K502, K503, K504, K505 }, \
+    { K600, K601, K602, K603, K604, K605 }, \
+    { K700, K701, K702, K703, K704, K705 }  \
+}
```
Comment (waffle87) [inline keyboards/2d4d/2d4d.h:50]: This can be migrated to `info.json` file and the `2d4d.[c,h]` files removed.

File: keyboards/2d4d/config.h [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    { B12, B13, B14, B15, A8, A15 }
+
+//A VE B kanalı çok önemli değil, eğer yanlış bağlarsan bunları yer değiştir.
+#ifdef ENCODER_ENABLE
+#define ENCODERS_PAD_A \
+    { A1 }
+#define ENCODERS_PAD_B \
+    { A0 }
+#define ENCODER_RESOLUTION 4
+#endif
```
Comment (waffle87) [inline keyboards/2d4d/config.h:34]: All this is required to be configured in `info.json`. Please see [PR checklist](https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs) for more info. (This file should be removed after doing so)

File: keyboards/2d4d/mcuconf.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+/*
+ * This file was auto-generated by:
+ *    `qmk chibios-confmigrate -i keyboards/zvecr/zv48/f401/mcuconf.h -r platforms/chibios/BLACKPILL_STM32_F401/configs/mcuconf.h`
+ */
+
+#pragma once
+
+#include_next <mcuconf.h>
+
+
```
Comment (waffle87) [inline keyboards/2d4d/mcuconf.h:26]: This file should be removed.

File: keyboards/2d4d/readme.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# 2d4d
+
+https://i.imgur.com/famGFG8.jpg
+
+https://i.imgur.com/IfoQUnc.jpg
+
+https://imgur.com/a/e9hCvfe
```
Comment (waffle87) [inline keyboards/2d4d/readme.md:7]:
```suggestion
![image](https://i.imgur.com/famGFG8h.jpeg)
```

File: keyboards/2d4d/readme.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+*Handwired %50 keyboard (48 switches), you can use 1 or 2 encoders instead of switch and vise versa*
+
+* Keyboard Maintainer: [mechMehmet](https://github.com/mechMehmet)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
```
Comment (waffle87) [inline keyboards/2d4d/readme.md:13]: These fields should be filled out accordingly.

File: keyboards/2d4d/info.json [hunk 7]
```diff
@@ -0,0 +1,74 @@
+{
+    "manufacturer": "mechMehmet",
+    "keyboard_name": "2d4d",
+    "maintainer": "mechMehmet",
+    "bootloader": "stm32-dfu",
+    "processor": "STM32F401",
```
Comment (waffle87) [inline keyboards/2d4d/info.json:6]:
```suggestion
    "processor": "STM32F401",
    "board": "BLACKPILL_STM32_F401",
    "features": {
        "bootmagic": true,
        "mousekey": true,
        "extrakey": true
    },
```

File: keyboards/2d4d/rules.mk [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+#
+BOOTMAGIC_ENABLE = yes # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes  # Mouse keys
+EXTRAKEY_ENABLE = yes  # Audio control and System control
+CONSOLE_ENABLE = no    # Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+NKRO_ENABLE = no      # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no  # Enable keyboard RGB underglow
+AUDIO_ENABLE = no      # Audio output
```
Comment (waffle87) [inline keyboards/2d4d/rules.mk:19]:
```suggestion
```

File: keyboards/2d4d/rules.mk [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+ENCODER_ENABLE = yes   # Enable encoder
```
Comment (waffle87) [inline keyboards/2d4d/rules.mk:20]: If encoders are enabled at the keyboard level, there should exist some functionality for them at the keyboard level as well, that way it works for QMK Configurator users. See [doc](https://docs.qmk.fm/#/feature_encoders?id=callbacks).

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by waffle87** [comment]
You have duplicated the original folder in the `handwired/` directory. Also, nearly none of the suggestions were applied...

**Comment by mechmehmet** [comment]
@waffle87 I really dont understand what suggestion you talking about? You are telling me the delete the <keyboard>.c,h files. I also changed so many files especially on info.json but you are telling me I didnt. you want me to change encoder to keyboard level but I cant if you look at the keymap how I use the encoder.

**Pull request closed without merging**